File: dev/customCPP/codegen/lib/myFun/myFun.cpp
//
// Non-Degree Granting Education License -- for use at non-degree
// granting, nonprofit, educational organizations only. Not for
// government, commercial, or other organizational use.
//
// myFun.cpp
//
// Code generation for function 'myFun'
//

// Include files
#include "myFun.h"
#include <cmath>

// Function Definitions
void myFun(const double params[4], double foo, double bar, double out1[2][2],
           double *out2)
{
  double myPar1;
  //  Vital we keep these variable names by way of explanation
  //  A comment here will explain the calculation
  myPar1 = (params[0] + params[3]) * (params[2] - foo);
  //  Another comment to keep here....
  //  Making the outputs is alse expained by variable names...
  out1[0][0] = myPar1;
  out1[1][0] = foo * bar;
  out1[0][1] = foo;
  out1[1][1] = bar;
  *out2 = std::sin(myPar1) * params[1];
}

// End of code generation (myFun.cpp)

File: dev/customCPP/myFun.m
function [out1, out2] = myFun(params, foo, bar)



% Vital we keep these variable names by way of explanation
carbon = params(1);
nitrogen = params(2);
oxygen = params(3);
hydrogen = params(4);

% A comment here will explain the calculation
myPar1 = (carbon + hydrogen) * (oxygen - foo);
myPar2 = foo * bar;

% Another comment to keep here....
row1 = [myPar1 myPar2];
row2 = [foo bar];

% Making the outputs is alse expained by variable names...
out1 = [row1 ; row2];
out2 = sin(myPar1) * nitrogen;

end

